Еще изменения для спейс подов

Miracler112 opened the pull request:
Новые изменения для спейс подов, надеюсь хоть немного, но они будут приближены к тому, что в будущем ими можно будет пользоваться на постоянной основе, где они не будут жутко багать и бесить всех остальных.

- Бласт доры спейс подов работают более менее нормально (нужны тесты)
- След за подом теперь двойной
- Появился фабрикатор для создания частей подов и некоторых модулей для него(боевые модули пока нельзя достать)
- Стрельба откатилась до прошлых значений, теперь прожектайлы не будут лететь сквозь стену
- При стрельбе под не поворачивается, что уменьшает шансы на попадание прожектайлов в себя
- Батарейки можно производить в pod fabricator

Commit:
Очередные фиксы подов, подкрутка значений и прочее

- Бласт доры спейс подов работают более менее нормально (нужны тесты)
- След за подом теперь двойной
- Появился фабрикатор для создания частей подов и некоторых модулей для него(боевые модули пока нельзя достать)
- Стрельба откатилась до прошлых значений, теперь прожектайлы не будут лететь сквозь стену
- При стрельбе под не поворачивается, что уменьшает шансы на попадание прожектайлов в себя

Commit:
Батарейки можно производить в pod fabricator

Commit:
Регестрация дефайна pod fabricator

Commit:
Новый эффект x64

Здесь новый эффект для того, что бы в будущем можно было без особых болезненных сложностей редактировать и изменять как угодно

Commit:
Новый файлик эффектов для 64x64

Commit:
Новая папка для бласт доров спейс подов

Commit:
Отедльные спрайты для бласт доров спейсподов

Это нужно для дальнейшего их изменения и модифицирования без вреда всему остальному, что к ним привязано из старого

Commit:
Сносим пустышку

LudwigVonChesterfield commented on code/game/objects/effects/effect_system.dm:
```diff
@@ -408,6 +420,38 @@ steam.start() -- spawns the effect
+/datum/effect/effect/system/ion_trail_follow_x64/set_up(atom/atom)
+	attach(atom)
+
+/datum/effect/effect/system/ion_trail_follow_x64/start() //Whoever is responsible for this abomination of code should become an hero
```
Я так понял в этот ПР лучше не смотреть?

Мне нравится как вы стырили вот этот урывок, даже с комментарием, при этом не прочитав комментария, и не обдумав что хотел сказать аффтар.

Miracler112 replied:
Ну почему же, можешь смотреть. Ревьювы в любом случае не помешают. Мне этот урывок нужен был для того, что бы в будущем потом на этой основе создать другие следы и не задеть при этом старые, в конце концов что эти, что старые в один тайл - не катят

LudwigVonChesterfield commented on code/modules/spacepods/spacepoddoor.dm:
```diff
@@ -56,21 +57,58 @@
+	var/obj/machinery/door/spacepoddoor/newmultidoor
+	if(type_bd == "4BV" || type_bd == "3BV")
+		var/turf/newturfmulti = get_step(src.loc, NORTH)
+		for(var/i = 0, i < width - 1, i++)
```
Красивый цикл.

Miracler112 replied:
Очень даже, но если такое не сойдет, то хорошо, поправлю

LudwigVonChesterfield replied:
Интересная особенность бьёнда, for(var/i in 0 to width - 1) работало бы быстрее, а ещё быстрее бы работало, если бы width - 1 было вынесено выше, иначе каждую итерацию цикла мы вычитаем единицу из ширины!

Miracler112 replied:
Ну вот по этому я это и поменяю. Так так видимо бьенд будет работать медленнее со всем этим, а раз так, то лучше сразу этого избегать

LudwigVonChesterfield commented on code/modules/spacepods/spacepoddoor.dm:
```diff
@@ -56,21 +57,58 @@
-/obj/machinery/doors/spacepoddoor/multi_tile/Destroy()
+/obj/machinery/door/spacepoddoor/multi_tile/Destroy()
+	for(var/obj/machinery/door/spacepoddoor/SD in src.multtile)
+		QDEL_NULL(SD)
```
А зачем нуллить ссылку SD отдельно, если она заннулится когда цикл пойдёт на следующую итерацию o.O

Miracler112 replied:
Во избежании всякого, но если обычный дел пройдет без всяких проблем, то поправлю тогда

Commit:
Правки замечаний людука

Commit:
Это мерзость, да

Но без неё в дальнейшем не сделать нормальные следы, так что хоть это и наглая и ужасная копипаста, но переработав её с нормальными эффектами будет смотреться хорошо

Miracler112 commented:
Усе
